test/general/implementation_details/contains_host_pointers.pass.cpp

```diff
@@ -0,0 +1,163 @@
+        static_assert(contains_host_pointer_v<decltype(zip_view)> == false);
+        static_assert(contains_host_pointer_on_any_layers_v<decltype(zip_view)> == true);
+    }
+
+    {
+        IntVector vec;
+
+        MinimalisticViewWithSubscription mrv(vec.begin(), vec.end());
+        auto all_view1 = std::ranges::views::all(mrv);
+
+        MinimalisticRangeForIntVec mr(vec.begin(), vec.end());
+        auto all_view2 = std::ranges::views::all(mr);
+
+        auto zip_view = oneapi::dpl::__ranges::make_zip_view(all_view1, all_view2);
+
+        static_assert(contains_host_pointer_v<decltype(all_view1)> == false);
+        static_assert(contains_host_pointer_v<decltype(all_view2)> == true);
+        static_assert(contains_host_pointer_v<decltype(zip_view)> == false);
+        static_assert(contains_host_pointer_on_any_layers_v<decltype(zip_view)> == true);
+    }
+
+    {
+        IntVector vec;
+
+        MinimalisticViewWithSubscription mrv(vec.begin(), vec.end());
+        auto all_view = std::ranges::views::all(mrv);
+
+        auto zip_view = oneapi::dpl::__ranges::make_zip_view(all_view, all_view);
+
+        static_assert(contains_host_pointer_v<decltype(all_view)> == false);
+        static_assert(contains_host_pointer_v<decltype(zip_view)> == false);
+        static_assert(contains_host_pointer_on_any_layers_v<decltype(zip_view)> == false);
+    }
+}
+
+// oneapi::dpl::__ranges::__contains_host_pointer functional with std::ranges::take_view
+void
+check_contains_host_pointer_in_take_view()
+{
+    IntVector vec;
+    MinimalisticRangeForIntVec mr(vec.begin(), vec.end());
+    auto all_view = std::ranges::views::all(mr);
+    auto taken_view = std::ranges::take_view(all_view, all_view.size());
+
+    static_assert(contains_host_pointer_v<decltype(all_view)> == true);
+    static_assert(contains_host_pointer_v<decltype(taken_view)> == false);
+    static_assert(contains_host_pointer_on_any_layers_v<decltype(taken_view)> == true);
+}
+
+// oneapi::dpl::__ranges::__contains_host_pointer functional with std::ranges::drop_view
+void
+check_contains_host_pointer_in_drop_view()
+{
+    IntVector vec;
+    MinimalisticRangeForIntVec mr(vec.begin(), vec.end());
+    auto all_view = std::ranges::views::all(mr);
+    auto dropped_view = std::ranges::drop_view(all_view, 0);
+
+    static_assert(contains_host_pointer_v<decltype(all_view)> == true);
+    static_assert(contains_host_pointer_v<decltype(dropped_view)> == false);
+    static_assert(contains_host_pointer_on_any_layers_v<decltype(dropped_view)> == true);
+}
+#endif // _ENABLE_STD_RANGES_TESTING && TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+    bool bProcessed = false;
+
+#if _ENABLE_STD_RANGES_TESTING && TEST_DPCPP_BACKEND_PRESENT
+
+    check_contains_host_pointer();
+    check_contains_host_pointer_in_zip_view();
+    check_contains_host_pointer_in_take_view();
+    check_contains_host_pointer_in_drop_view();
```

Comment:
One more suggestion is to add a direct case with `std::ranges::ref_view`. It should also verify a new static_assert from `__require_access`, checking the top-level range.